### PR TITLE
Fixes #10375 - entering edit form deletes attributes

### PR DESCRIPTION
**Pull request by lzap**
We simply deleted attributes on edit form render. I added a comment explaining
the method and how it should be used.

### CODE REVIEW

File: app/controllers/discovered_hosts_controller.rb [hunk 1]
```diff
@@ -60,7 +60,7 @@ def destroy
 
   def edit
     Host.transaction do
-      @host = ::ForemanDiscovery::HostConverter.to_managed(@host) unless @host.nil?
+      @host = ::ForemanDiscovery::HostConverter.to_managed(@host, true, false, false) unless @host.nil?
```
Comment (orrabin) [inline app/controllers/discovered_hosts_controller.rb]: is this the only place we shouldn't delete the attributes?
we have `to_managed` in `perform_auto_provision` in `discovered_extensions.rb`, if the auto_provision fails what happens?

**Comment by orrabin** [comment]
I haven't has a chance to test with failing auto provision, I tested with provision (successful and unsuccessful) and successful auto privision and it works as expected.
Also, nitpick - can you change  the commit message to - entering edit form doesn't delete attributes?

File: app/controllers/discovered_hosts_controller.rb (hunk 1 above)
Comment (lzap) [inline app/controllers/discovered_hosts_controller.rb]: Yes, this is the only place.

**Comment by lzap** [comment]
Re-phrased.

**Comment by ohadlevy** [comment]
missing tests?

**Comment by orrabin** [comment]
@lzap I was able to delete the attributes before a successful provision by:
1. click provision
2. try to save
3. cancel
just entering the edit form or clicking cancel right away didn't delete the attributes

**Comment by lzap** [comment]
@orrabin as per our call from this morning, you are actually working on a test to reproduce this, right? I will wait then...

**Comment by orrabin** [comment]
@lzap, not this one but we need to add a test for this too

**Comment by lzap** [comment]
Test added.

File: app/controllers/discovered_hosts_controller.rb (hunk 1 above)
Comment (lzap) [inline app/controllers/discovered_hosts_controller.rb]: @orrabin here I changed the 3rd param to `false`

**Comment by lzap** [comment]
Pushed

File: app/models/host/managed_extensions.rb [hunk 2]
```diff
@@ -24,4 +25,9 @@ def delReboot
     # last action in the queue.
   end
 
+  def delete_discovery_attribute_set
+    return if new_record?
+    DiscoveryAttributeSet.destroy_all(:host_id => host_id) if type_changed?
```
Comment (orrabin) [inline app/models/host/managed_extensions.rb]: maybe add if old was discovered host?

**Comment by orrabin** [comment]
tested locally and it looks good, waiting for Jenkins

**Comment by orrabin** [comment]
looks like are tests are not so reliable right now: http://projects.theforeman.org/issues/10548
I tested manually and we need to rewrite the tests and check this but for now merging as is

**Comment by orrabin** [comment]
merged as 009bb13e093bef968461ca4121e97fbfa1d2d370, thanks @lzap!

**Pull request closed without merging**